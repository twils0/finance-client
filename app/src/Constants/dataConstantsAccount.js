export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
