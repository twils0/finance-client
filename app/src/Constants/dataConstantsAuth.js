export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
