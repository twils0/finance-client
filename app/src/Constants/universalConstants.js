import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
