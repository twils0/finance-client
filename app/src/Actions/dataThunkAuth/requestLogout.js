import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import handleErrorCatch from '../../handleErrorCatch';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAuth';
import { formNames, buttonNames, buttonTexts } from '../../Constants/uiConstantsAccount';
import { setAuthStatus, resetAuthState } from '../dataActionsAuth';
import { resetAWSState } from '../dataActionsAWS';
import { resetAccountState } from '../dataActionsAccount';
import { resetWatchlistState } from '../dataActionsWatchlist';
import { setCurrentForm, setButtonText, setButtonVisible } from '../uiActionsAccount';

const requestLogout = () => async (dispatch, getState) => {
  const state = getState();

  if (state.data.auth.status[statusNames.LOGOUT].status !== requestStatusTypes.LOADING) {
    dispatch(
      setAuthStatus({
        id: statusNames.LOGOUT,
        status: requestStatusTypes.LOADING,
      }),
    );

    if (state.ui.internal.account.forms.current !== formNames.PROFILE) {
      dispatch(setCurrentForm({ current: formNames.PROFILE }));
    }

    if (state.ui.internal.account.buttons[buttonNames.FIRST].text !== buttonTexts.EDIT) {
      dispatch(setButtonText({ id: buttonNames.FIRST, text: buttonTexts.EDIT }));
    }

    if (state.ui.internal.account.buttons[buttonNames.SECOND].visible) {
      dispatch(setButtonVisible({ id: buttonNames.SECOND, visible: false }));
    }

    try {
      await Auth.signOut();
    } catch (errorCatch) {
      const error = handleErrorCatch(errorCatch);

      dispatch(
        setAuthStatus({
          id: statusNames.LOGOUT,
          status: requestStatusTypes.ERROR,
        }),
      );

      raven.captureException(error, {
        logger: 'requestLogout',
      });

      return Promise.reject(error);
    }

    window.sessionStorage.removeSecurity('sessionTime');

    const intervalId = window.sessionStorage.getSecurity('interval');

    if (intervalId) {
      window.clearInterval(intervalId);

      window.sessionStorage.removeSecurity('interval');
    }

    dispatch(resetAWSState());
    dispatch(resetAccountState());
    dispatch(resetWatchlistState());
    dispatch(resetAuthState());

    raven.setUserContext();
  }

  return null;
};

export default requestLogout;
