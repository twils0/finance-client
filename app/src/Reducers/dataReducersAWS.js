import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes } from '../Constants/dataConstantsAWS';

export const initialState = {
  status: requestStatusTypes.IDLE,
  user: null,
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_STATUS:
      return action.payload.status;
    case actionTypes.RESET_AWS_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const user = (state = initialState.user, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_USER:
      return action.payload.user;
    case actionTypes.RESET_AWS_STATE:
      return initialState.user;
    default:
      return state;
  }
};

const dataReducersAWS = combineReducers({
  status,
  user,
});

export default dataReducersAWS;
