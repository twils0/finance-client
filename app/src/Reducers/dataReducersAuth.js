import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
