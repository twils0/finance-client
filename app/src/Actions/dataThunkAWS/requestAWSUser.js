import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import handleErrorCatch from '../../handleErrorCatch';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { setAWSStatus, setAWSUser } from '../dataActionsAWS';
import requestLogout from '../dataThunkAuth/requestLogout';

import { demo } from '../../../../mode.config.json';

const requestAWSUser = () => async (dispatch, getState) => {
  const state = getState();

  if (state.data.aws.status !== requestStatusTypes.LOADING) {
    dispatch(setAWSStatus({ status: requestStatusTypes.LOADING }));

    let user = null;

    try {
      if (!demo) {
        user = await Auth.currentUserPoolUser();
      }
    } catch (errorCatch) {
      const error = handleErrorCatch(errorCatch);

      dispatch(setAWSStatus({ status: requestStatusTypes.ERROR }));

      if (
        (typeof error === 'string'
          && (error === 'not authenticated' || error === 'No current user in userPool'))
        || (typeof error === 'object' && error && error.code === 'NotAuthorizedException')
      ) {
        dispatch(requestLogout());

        return null;
      }

      if (!demo) {
        raven.captureException(error, {
          logger: 'requestAWSUser',
        });
      }

      return Promise.reject(error);
    }

    if (!demo) {
      dispatch(setAWSUser({ user }));
    }
    dispatch(setAWSStatus({ status: requestStatusTypes.SUCCESS }));

    return { user };
  }

  return null;
};

export default requestAWSUser;
