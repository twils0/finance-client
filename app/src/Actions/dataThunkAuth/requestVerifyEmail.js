import axios from 'axios';
import raven from 'raven-js';

import { requestStatusTypes, URLs, axiosConfig } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAuth';
import requestAWSUser from '../dataThunkAWS/requestAWSUser';
import requestLogout from './requestLogout';
import { setAuthStatus } from '../dataActionsAuth';

import handleErrorCatch from '../../handleErrorCatch';

import { demo } from '../../../../mode.config.json';

const requestVerifyEmail = () => async (dispatch, getState) => {
  const state = getState();

  if (state.data.auth.status[statusNames.VERIFY_EMAIL].status !== requestStatusTypes.LOADING) {
    dispatch(
      setAuthStatus({
        id: statusNames.VERIFY_EMAIL,
        status: requestStatusTypes.LOADING,
      }),
    );

    let { user } = state.data.aws;

    try {
      if (!user) {
        ({ user } = await dispatch(requestAWSUser()));
      }

      if (!demo) {
        const idToken = user.signInUserSession.idToken.jwtToken;
        const accessToken = user.signInUserSession.accessToken.jwtToken;

        await axios.post(
          URLs.EMAILS,
          {},
          {
            headers: {
              Authorization: idToken,
            },
            params: {
              accessToken,
            },
            ...axiosConfig.DB,
          },
        );
      }
    } catch (errorCatch) {
      const error = handleErrorCatch(errorCatch);

      dispatch(
        setAuthStatus({
          id: statusNames.VERIFY_EMAIL,
          status: requestStatusTypes.ERROR,
        }),
      );

      if (typeof error === 'object' && error && error.code === 'NotAuthorizedException') {
        dispatch(requestLogout());

        return null;
      }

      if (!demo) {
        raven.captureException(error, {
          logger: 'requestVerifyEmail',
        });
      }

      return Promise.reject(error);
    }

    dispatch(
      setAuthStatus({
        id: statusNames.VERIFY_EMAIL,
        status: requestStatusTypes.SUCCESS,
      }),
    );
  }

  return null;
};

export default requestVerifyEmail;
