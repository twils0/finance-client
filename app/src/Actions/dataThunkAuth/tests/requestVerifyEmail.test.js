import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios';

import { initialState as initialStateAuth } from '../../../Reducers/dataReducersAuth';
import { initialState as initialStateAWS } from '../../../Reducers/dataReducersAWS';
import { requestStatusTypes, URLs, axiosConfig } from '../../../Constants/universalConstants';
import { actionTypes as actionTypesAuth, statusNames } from '../../../Constants/dataConstantsAuth';
import requestLogout from '../../dataThunkAuth/requestLogout';
import requestAWSUser from '../../dataThunkAWS/requestAWSUser';
import requestVerifyEmail from '../requestVerifyEmail';

jest.mock('axios', () => ({
  post: jest.fn(),
}));
jest.mock('../../dataThunkAuth/requestLogout', () => jest.fn());
jest.mock('../../dataThunkAWS/requestAWSUser', () => jest.fn());

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const userObject = {
  user: {
    signInUserSession: {
      idToken: { jwtToken: 'testIdToken' },
      accessToken: { jwtToken: 'testAccessToken' },
    },
  },
};
const stripeResponse = {
  data: {
    body: {
      nameOnCard: 'testName',
      promoCode: 'testPromo',
      promoCodeValid: true,
    },
  },
};
const idToken = userObject.user.signInUserSession.idToken.jwtToken;
const accessToken = userObject.user.signInUserSession.accessToken.jwtToken;
const mockAxiosConfig = {
  headers: {
    Authorization: idToken,
  },
  params: {
    accessToken,
  },
  ...axiosConfig.DB,
};

describe('dataThunkAuth', () => {
  describe('requestVerifyEmail', () => {
    afterEach(() => {
      requestAWSUser.mockReset();
      axios.post.mockReset();
      requestLogout.mockReset();
    });

    it("fails and throws an error when missing 'match' key in payload", async () => {
      const store = mockStore({ data: { auth: initialStateAuth } });

      const emptyPayload = {};

      try {
        await store.dispatch(requestVerifyEmail(emptyPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      expect(requestAWSUser).not.toBeCalled();
      expect(axios.post).not.toBeCalled();
      expect(requestLogout).not.toBeCalled();
    });

    it("calls requestAWSUser when missing 'user' object in state", async () => {
      const store = mockStore({ data: { auth: initialStateAuth, aws: initialStateAWS } });

      requestAWSUser.mockReturnValue(() => Promise.resolve(userObject));
      axios.post.mockReturnValue(Promise.resolve(stripeResponse));

      await store.dispatch(requestVerifyEmail());

      expect(requestAWSUser).toBeCalled();
      expect(axios.post).toBeCalledWith(URLs.EMAILS, {}, mockAxiosConfig);
      expect(requestLogout).not.toBeCalled();
    });

    it('fails and returns promise reject when requestAWSUser throws an error', async () => {
      const store = mockStore({ data: { auth: initialStateAuth, aws: initialStateAWS } });

      const error = { code: 'testError', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      requestAWSUser.mockReturnValue(() => Promise.reject(error));

      try {
        await store.dispatch(requestVerifyEmail());
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(requestAWSUser).toBeCalled();
      expect(axios.post).not.toBeCalled();
      expect(requestLogout).not.toBeCalled();
    });

    it("calls requestLogout when axios throws 'NotAuthorizedException' error", async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
      stateBeforeAWS.user = userObject.user;

      const store = mockStore({ data: { auth: initialStateAuth, aws: stateBeforeAWS } });

      const error = { code: 'NotAuthorizedException', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      axios.post.mockReturnValue(Promise.reject(error));
      requestLogout.mockReturnValue(() => null);

      try {
        await store.dispatch(requestVerifyEmail());
      } catch (errorCatch) {
        expect(errorCatch).toEqual(null);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(requestAWSUser).not.toBeCalled();
      expect(axios.post).toBeCalledWith(URLs.EMAILS, {}, mockAxiosConfig);
      expect(requestLogout).toBeCalled();
    });

    it('fails and returns promise reject when axios throws an error', async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
      stateBeforeAWS.user = userObject.user;

      const store = mockStore({ data: { auth: initialStateAuth, aws: stateBeforeAWS } });

      const error = { code: 'testError', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      axios.post.mockReturnValue(Promise.reject(error));

      try {
        await store.dispatch(requestVerifyEmail());
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(requestAWSUser).not.toBeCalled();
      expect(axios.post).toBeCalledWith(URLs.EMAILS, {}, mockAxiosConfig);
      expect(requestLogout).not.toBeCalled();
    });

    it('creates the correct actions with the correct payloads', async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));

      stateBeforeAWS.user = userObject.user;
      const store = mockStore({ data: { auth: initialStateAuth, aws: stateBeforeAWS } });

      const expectedActions = [
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAuth.SET_AUTH_STATUS,
          payload: {
            id: statusNames.VERIFY_EMAIL,
            status: requestStatusTypes.SUCCESS,
          },
        },
      ];

      axios.post.mockReturnValue(Promise.resolve(null));

      const result = await store.dispatch(requestVerifyEmail());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(requestAWSUser).not.toBeCalled();
      expect(axios.post).toBeCalledWith(URLs.EMAILS, {}, mockAxiosConfig);
      expect(requestLogout).not.toBeCalled();
    });
  });
});
