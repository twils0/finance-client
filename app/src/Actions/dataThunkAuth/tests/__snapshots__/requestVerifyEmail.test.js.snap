// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataThunkAuth requestVerifyEmail fails and throws an error when missing 'match' key in payload 1`] = `[TypeError: Cannot read property 'user' of undefined]`;
